src/ansys/fluent/core/launcher/launcher_arguments.py

```diff
@@ -0,0 +1,346 @@
+from enum import Enum
+import os
+from pathlib import Path
+import platform
+import subprocess
+import tempfile
+from typing import Any, Dict, Optional, Union
+
+from ansys.fluent.core.exceptions import DisallowedValuesError
+from ansys.fluent.core.fluent_connection import FluentConnection, PortNotProvided
+from ansys.fluent.core.launcher.custom_exceptions import (
+    DockerContainerLaunchNotSupported,
+    InvalidPassword,
+    IpPortNotProvided,
+    UnexpectedKeywordArgument,
+)
+from ansys.fluent.core.launcher.launcher_utils import check_docker_support, logger
+from ansys.fluent.core.session import _parse_server_info_file
+from ansys.fluent.core.session_meshing import Meshing
+from ansys.fluent.core.session_pure_meshing import PureMeshing
+from ansys.fluent.core.session_solver import Solver
+from ansys.fluent.core.session_solver_icing import SolverIcing
+from ansys.fluent.core.utils.fluent_version import FluentVersion
+from ansys.fluent.core.utils.networking import find_remoting_ip
+import ansys.platform.instancemanagement as pypim
+
+
+class LaunchMode(Enum):
+    """An enumeration over supported Fluent launch modes."""
+
+    STANDALONE = 1
+    PIM = 2
+    CONTAINER = 3
+    SLURM = 4
+
+
+class FluentMode(Enum):
+    """An enumeration over supported Fluent modes."""
+
+    MESHING_MODE = (Meshing, "meshing")
+    PURE_MESHING_MODE = (PureMeshing, "pure-meshing")
+    SOLVER = (Solver, "solver")
+    SOLVER_ICING = (SolverIcing, "solver-icing")
+
+    @staticmethod
+    def get_mode(mode: str) -> "FluentMode":
+        """Returns the FluentMode based on the provided mode string.
```

Comment:
```suggestion
        """Get the Fluent mode based on the provided mode string.
```
Docstrings should use plain English whenever possible.